test/functional/sqlsrv/sqlsrv_ae_output_param_sqltype_string.phpt

```diff
@@ -0,0 +1,123 @@
+                     "varchar(max)" => array( "SQLSRV_SQLTYPE_CHAR(5)", "SQLSRV_SQLTYPE_VARCHAR", "SQLSRV_SQLTYPE_NCHAR(5)", "SQLSRV_SQLTYPE_NVARCHAR", "SQLSRV_SQLTYPE_DECIMAL", "SQLSRV_SQLTYPE_NUMERIC", "SQLSRV_SQLTYPE_NTEXT", "SQLSRV_SQLTYPE_TEXT", "SQLSRV_SQLTYPE_XML" ),
+                     "nchar(5)" => array( "SQLSRV_SQLTYPE_CHAR(5)", "SQLSRV_SQLTYPE_VARCHAR", "SQLSRV_SQLTYPE_NCHAR(5)", "SQLSRV_SQLTYPE_NVARCHAR", "SQLSRV_SQLTYPE_DECIMAL", "SQLSRV_SQLTYPE_NUMERIC", "SQLSRV_SQLTYPE_NTEXT", "SQLSRV_SQLTYPE_TEXT", "SQLSRV_SQLTYPE_XML" ),
+                     "nvarchar(max)" => array( "SQLSRV_SQLTYPE_CHAR(5)", "SQLSRV_SQLTYPE_VARCHAR", "SQLSRV_SQLTYPE_NCHAR(5)", "SQLSRV_SQLTYPE_NVARCHAR", "SQLSRV_SQLTYPE_DECIMAL", "SQLSRV_SQLTYPE_NUMERIC", "SQLSRV_SQLTYPE_NTEXT", "SQLSRV_SQLTYPE_TEXT", "SQLSRV_SQLTYPE_XML" ));
+
+$conn = AE\connect();
+	
+foreach ($dataTypes as $dataType) {
+    echo "\nTesting $dataType: \n";
+    $success = true;
+
+    // create table
+    $tbname = GetTempTableName("", false);
+    $colMetaArr = array( new AE\ColumnMeta($dataType, "c_det"), new AE\ColumnMeta($dataType, "c_rand", null, false));
+    AE\createTable($conn, $tbname, $colMetaArr);
+	
+    // insert a row
+    $inputValues = array_slice(${explode("(", $dataType)[0] . "_params"}, 1, 2);
+    $r;
+    $stmt = AE\insertRow($conn, $tbname, array( $colMetaArr[0]->colName => $inputValues[0], $colMetaArr[1]->colName => $inputValues[1] ), $r);
+    if ($r === false) {
+        is_incompatible_types_error($dataType, "default type");
+    }
+	
+    // Create a Store Procedure
+    $spname = 'selectAllColumns';
+    $spSql = "CREATE PROCEDURE $spname (@c_det $dataType OUTPUT, @c_rand $dataType OUTPUT ) AS SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname";
+    sqlsrv_query($conn, $spSql);
+
+    // test each SQLSRV_SQLTYPE_ constants
+    foreach ($sqlTypes as $sqlType) {
+        if (!AE\isColEncrypted()) {
+            $isCompatible = false;
+            foreach ($compatList[$dataType] as $compatType) {
+                if (stripos($compatType, $sqlType) !== false) {
+                    $isCompatible = true;
+                }
+            }
+            // 22018 is the SQLSTATE for any incompatible conversion errors
+            if ($isCompatible && sqlsrv_errors()[0]['SQLSTATE'] == 22018) {
+                echo "$sqlType should be compatible with $dataType\n";
+                $success = false;
+            }
+        } else {
+            // always encrypted only allow sqlType that is identical to the encrypted column datatype
+            if (stripos("SQLSRV_SQLTYPE_" . $dataType, $sqlType) !== false) {
+		        // insert a row
+                $inputValues = array_slice(${explode("(", $dataType)[0] . "_params"}, 1, 2);
+                $sqlType = get_default_size_prec($sqlType);
+                $inputs = array(new AE\BindParamOption($inputValues[0], null, null, $sqlType), new AE\BindParamOption($inputValues[1], null, null, $sqlType));

```

Comment:
$inputValues and $inputs are not actually used, you can remove them. Replace `count($inputs)` at line 67 with 2 representing 2 parameters in the SQL query.